[gtest] build failure while using GCC 11 (trunk)

**Host Environment**
- OS: Linux 5.8.18-1-MANJARO
- Compiler: g++ (GCC) 11.0.0 20201227 (experimental)

**To Reproduce**
`./vcpkg install gtest`

**Failure logs** 
```
Computing installation plan...
The following packages will be built and installed:
    gtest[core]:x64-linux
Detecting compiler hash for triplet x64-linux...
Starting package 1/1: gtest:x64-linux
Building package gtest[core]:x64-linux...
Could not locate cached archive: /home/twice/.cache/vcpkg/archives/1a/1ac447f4d814709eb219cbfb117dd4a60277c43d.zip
-- Using cached /home/twice/vcpkg/downloads/google-googletest-release-1.10.0.tar.gz
-- Cleaning sources at /home/twice/vcpkg/buildtrees/gtest/src/ase-1.10.0-7231afa423.clean. Use --editable to skip cleaning for the packages you specify.

-- Extracting source /home/twice/vcpkg/downloads/google-googletest-release-1.10.0.tar.gz
-- Applying patch 0002-Fix-z7-override.patch
-- Applying patch fix-main-lib-path.patch
-- Using source at /home/twice/vcpkg/buildtrees/gtest/src/ase-1.10.0-7231afa423.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:142 (message):
    Command failed: /home/twice/vcpkg/downloads/tools/cmake-3.18.4-linux/cmake-3.18.4-Linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /home/twice/vcpkg/buildtrees/gtest/x64-linux-dbg
    See logs for more information:
      /home/twice/vcpkg/buildtrees/gtest/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:93 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/gtest/portfile.cmake:28 (vcpkg_install_cmake)
  scripts/ports.cmake:135 (include)


Error: Building package gtest:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gtest:x64-linux
  Vcpkg version: 2020.06.15-unknownhash

Additionally, attach any relevant sections from the log files above.
```
[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/5806435/install-x64-linux-dbg-out.log)

**Additional context**
The build error is caused by a new warning (`something may be used uninitialized`) in GCC 11 and `-Werror` in CMake configuration of `gtest`.
Master branch of `gtest` has [fixed this compile error](https://github.com/google/googletest/blob/master/googletest/src/gtest-death-test.cc#L1291), but we cannot expect any releases due to [Abseil's Live-at-Head philosophy](https://github.com/google/googletest/issues/3124).
So how about adding a patch for it?

--- a/port_versions/baseline.json
+++ b/port_versions/baseline.json
@@ -2270,7 +2270,7 @@
     },
     "gtest": {
       "baseline": "1.10.0",
-      "port-version": 2
+      "port-version": 3
     },
     "gtk": {
       "baseline": "3.22.19-4",

--- a/port_versions/g-/gtest.json
+++ b/port_versions/g-/gtest.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e94f8c09a8fa6ed7fc618734d412878d83069bfb",
+      "version-string": "1.10.0",
+      "port-version": 3
+    },
+    {
       "git-tree": "88de073e86bcae80206fca0ff4f4f8e6c165f43a",
       "version-string": "1.10.0",
       "port-version": 2

--- a/ports/gtest/CONTROL
+++ b/ports/gtest/CONTROL
@@ -1,5 +1,5 @@
 Source: gtest
 Version: 1.10.0
-Port-Version: 2
+Port-Version: 3
 Homepage: https://github.com/google/googletest
 Description: GoogleTest and GoogleMock testing frameworks.

--- a/ports/gtest/fix-build-failure-in-gcc-11.patch
+++ b/ports/gtest/fix-build-failure-in-gcc-11.patch
@@ -1,0 +1,22 @@
+diff --git a/googletest/src/gtest-death-test.cc b/googletest/src/gtest-death-test.cc
+index da09a1cf..43cbd78a 100644
+--- a/googletest/src/gtest-death-test.cc
++++ b/googletest/src/gtest-death-test.cc
+@@ -1288,7 +1288,7 @@ static void StackLowerThanAddress(const void* ptr,
+ // making comparison result unpredictable.
+ GTEST_ATTRIBUTE_NO_SANITIZE_HWADDRESS_
+ static void StackLowerThanAddress(const void* ptr, bool* result) {
+-  int dummy;
++  int dummy = 0;
+   *result = (&dummy < ptr);
+ }
+ 
+@@ -1296,7 +1296,7 @@ static void StackLowerThanAddress(const void* ptr, bool* result) {
+ GTEST_ATTRIBUTE_NO_SANITIZE_ADDRESS_
+ GTEST_ATTRIBUTE_NO_SANITIZE_HWADDRESS_
+ static bool StackGrowsDown() {
+-  int dummy;
++  int dummy = 0;
+   bool result;
+   StackLowerThanAddress(&dummy, &result);
+   return result;

--- a/ports/gtest/portfile.cmake
+++ b/ports/gtest/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_from_github(
     PATCHES
         0002-Fix-z7-override.patch
         fix-main-lib-path.patch
+        fix-build-failure-in-gcc-11.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "dynamic" GTEST_FORCE_SHARED_CRT)

